File: src/main/java/com/giant/common/util/CookieUtil.java
package com.giant.common.util;

import jakarta.servlet.http.HttpServletResponse;
import org.springframework.http.HttpHeaders;
import org.springframework.stereotype.Component;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseCookie;

@Component
public class CookieUtil {
    private final boolean isSecured;

    public CookieUtil(@Value("${spring.profiles.active}") String activeProfile) {
        isSecured = !"local".equals(activeProfile);
    }

    public void addCookie(
            HttpServletResponse response,
            String name,
            String value,
            long maxAge,
            boolean isHttpOnly
    ) {
        ResponseCookie cookie = ResponseCookie.from(name, value)
                .path("/")
                .httpOnly(isHttpOnly)
                .maxAge(maxAge)
                .sameSite("Lax")
                .secure(isSecured)
                .build();
        response.addHeader(HttpHeaders.SET_COOKIE, cookie.toString());
    }

    public void removeCookie(HttpServletResponse response, String name, boolean isHttpOnly) {
        addCookie(response, name, "", 0, isHttpOnly);
    }
}

File: src/main/kotlin/com/giant/common/util/CookieUtil.kt
package com.giant.common.util

import jakarta.servlet.http.HttpServletResponse
import org.springframework.http.HttpHeaders
import org.springframework.http.ResponseCookie

fun addCookie(
    response: HttpServletResponse,
    name:String,
    value:String,
    isHttpOnly: Boolean,
    maxAgeInSeconds:Long
)  {
    val cookie = ResponseCookie.from(name, value)
        .path("/")
        .httpOnly(isHttpOnly)
        .secure(true)
        .maxAge(maxAgeInSeconds)
        .sameSite("Lax")
        .build()

    response.addHeader(HttpHeaders.SET_COOKIE, cookie.toString())
}

fun removeCookie(
    response: HttpServletResponse,
    name:String,
    isHttpOnly: Boolean,
) {
    val cookie =  ResponseCookie.from(name, "")
        .path("/")
        .httpOnly(isHttpOnly)
        .secure(true)
        .maxAge(0)
        .sameSite("Lax")
        .build()

    response.addHeader(HttpHeaders.SET_COOKIE, cookie.toString())
}